releasenotes/notes/add-backend-v2-ce84c976fb13b038.yaml

```diff
@@ -0,0 +1,87 @@
+    changes both the user access pattern for properties of the backend (like
+    number of qubits, etc) and how the backend represents its constraints
+    to the transpiler. The execution of circuits (via the
+    :meth:`~qiskit.providers.BackendV2.run` method) remains unchanged. With
+    a :class:`~qiskit.providers.BackendV2` backend instead of having a separate
+    :meth:`~qiskit.providers.BackendV1.configuration`,
+    :meth:`~qiskit.providers.BackendV1.properties`, and
+    :meth:`~qiskit.providers.BackendV1.defaults` methods that construct
+    :class:`~qiskit.providers.models.BackendConfiguration`, 
+    :class:`~qiskit.providers.models.BackendProperties`, and
+    :class:`~qiskit.providers.models.PulseDefaults` objects respectively,
+    like in the :class:`~qiskit.providers.BackendV1` interface, the attributes
+    contained in those output objects are accessible directly as attributes of
+    the :class:`~qiskit.providers.BackendV2` object. For example, to get the
+    number of qubits for a backend with :class:`~qiskit.providers.BackendV1`
+    you would do::
+
+      num_qubits = backend.configuration().n_qubits
+
+    while with :class:`~qiskit.providers.BackendV2` it is::
+
+      num_qubits = backend.num_qubits
+
+    The other change around this is that the number of attributes exposed in
+    the abstract :class:`~qiskit.providers.BackendV2` class is designed to be
+    a hardware/vendor agnostic set of the required or optional fields that the
+    rest of Qiskit can use today with any backend. Anything can be added
+    to a particular implementation of :class:`~qiskit.providers.BackendV2`
+    as a custom addition for that particular backend. But it's not something
+    that is expected to be exposed by all backends.
+
+    The other critical change is primarily important for provider authors is
+    how a :class:`~qiskit.providers.BackendV2` exposes the properties of
+    a particular backend to the transpiler. With
+    :class:`~qiskit.providers.BackendV2` this is done via a
+    :class:`~qiskit.transpiler.Target` object. The
+    :class:`~qiskit.transpiler.Target`, which is exposed via the
+    :attr:`~qiskit.providers.BackendV2.target` attribute, is used to represent
+    the set of constraints for running circuits on a particular backend. It
+    contains the subset of information previously exposed by the
+    :class:`~qiskit.providers.models.BackendConfiguration`,
+    :class:`~qiskit.providers.models.BackendProperties`, and
+    :class:`~qiskit.providers.models.PulseDefaults` classes which the transpiler
+    can actively use. When migrating a provider to use
+    :class:`~qiskit.providers.BackendV2` (or when creating a new provider
+    package) the construction of backend objects will primarily be around
+    creating a :class:`~qiskit.transpiler.Target` object for the backend.
+  - |
+    Added a new :class:`~qiskit.transpiler.Target` class to the
```

Comment:
Does the `Target`, being a class built and returned by a `BackendV2`, have any additional versioning/support constraints? That is, if `Backend` is explicitly versioned class, should `Target` be also?

Reply:
We discussed this offline, but leaving a comment here. The `Target` class added here is basically equivalent to V0, if we need to change it in a backwards incompatible manner (adding new required arguments etc), we can add explicit versioning to it for a `TargetV1`m look at doing mixins for the additional fields, etc. (this is what I was describing as part of my comment here: https://github.com/Qiskit/qiskit-terra/pull/5885#discussion_r752387719 ). So the answer is yes we should be versioning it in the future, but just we're not doing it yet (to leave open flexibility of using different mechanism than an explicit version).